geopandas/array.py

```diff
@@ -828,14 +828,27 @@ def estimate_utm_crs(self, datum_name="WGS 84"):
             raise RuntimeError("crs must be set to estimate UTM CRS.")
 
         minx, miny, maxx, maxy = self.total_bounds
+        x_center = None
+        y_center = None
         # ensure using geographic coordinates
         if not self.crs.is_geographic:
-            lon, lat = Transformer.from_crs(
-                self.crs, "EPSG:4326", always_xy=True
-            ).transform((minx, maxx, minx, maxx), (miny, miny, maxy, maxy))
-            x_center = np.mean(lon)
-            y_center = np.mean(lat)
-        else:
+            transformer = Transformer.from_crs(self.crs, "EPSG:4326", always_xy=True)
+            if compat.PYPROJ_GE_31:
+                minx, miny, maxx, maxy = transformer.transform_bounds(
+                    minx, miny, maxx, maxy
+                )
+                if minx > maxx:
+                    # crossed the antimeridian
+                    minx = -180
+                    maxx = -180
```

Comment:
Just for clarification, this is here to ensure that the mean below is -180 hence we assign the correct zone, right? It is a bit cryptic at first sight.

Reply:
Correct. Updated the logic and added some notes to helpfully make it clearer.